core/src/main/java/io/questdb/client/impl/ConfStringParser.java

```diff
@@ -0,0 +1,172 @@
+                return -1;
+            } else if (invalidIdentifierChar(c)) {
+                output.put("key must be consist of alpha-numerical ascii characters and underscore, not '").put(c).put("' at position ").put(pos);
+                return -1;
+            }
+
+        }
+        output.clear();
+        output.put("incomplete key-value pair before end of input at position ").put(pos);
+        return -1;
+    }
+
+    /**
+     * Parses schema name from input. Schema name must start with schema type, e.g. http::.
+     * <p>
+     * This is the starting point for all configuration parsing. It returns a position handler
+     * that can be used to parse the rest of the configuration.
+     * <p>
+     * When parsing configuration, the parser will return -1 if it encounters an error. In this
+     * case, the error message will be written to the output sink.
+     *
+     * @param input  input
+     * @param output output
+     * @return position handler, or negative value if error
+     */
+    public static int of(CharSequence input, StringSink output) {
+        output.clear();
+        char lastChar = 0;
+        for (int i = 0, n = input.length(); i < n; i++) {
+            char c = input.charAt(i);
+            if (lastChar == ':') {
+                if (c == ':') {
+                    if (i == 1) {
+                        output.put("schema is empty");
+                        return -1;
+                    }
+                    if (i == n - 1) {
+                        output.put("missing trailing semicolon at position ").put(i);
+                        return -1;
+                    }
+                    Chars.toLowerCase(input, 0, i - 1, output);
+                    return i + 1;
+                } else {
+                    output.put("bad separator, expected '::' got ':").put(c).put("' at position ").put(i - 1);
+                    return -1;
+                }
+            } else if (c == ':') {
+                lastChar = c;
+            } else if (invalidIdentifierChar(c)) {
+                output.put("bad separator, expected ':' got '").put(c).put("' at position ").put(i);
+                return -1;
+            } else {
+                lastChar = c;
+            }
+        }
+        output.put("schema name must start with schema type, e.g. http::");
+        return -1;
+    }
+
+    /**
+     * Parse value from input and write it to output.
+     * <p>
+     * Call this method only after a previous call to {@link #nextKey(CharSequence, int, StringSink)} returned
+     * a non-negative value.
+     * <p>
+     * This method will return -1 if it encounters an error.
+     *
+     * @param input  input
+     * @param pos    position handler
+     * @param output output
+     * @return next position handler, or negative value if error
+     */
+    public static int value(CharSequence input, int pos, StringSink output) {
```

Comment:
missing logic to error if there are control characters.
I.e. any in the `0x00..=0x1f` and `0x7f..=0x9f` ranges.

See: https://github.com/questdb/questdb-confstr-rs/blob/be71c4e83a64ecd9517c75499527d3bc186532c4/questdb-confstr/src/lib.rs#L217